### PR TITLE
fix writing latency to bigquery in deploy check

### PR DESCRIPTION
**Pull request by nkubala**
also, handle error correctly if the deployment fails so we don't leak deployed versions.

### CODE REVIEW

**Comment by nkubala** [comment]
cc @liyanhui1228

File: integration_tests/testsuite/deploy_app.py [hunk 1]
```diff
@@ -72,9 +72,9 @@ def _record_latency_to_bigquery(deploy_latency, language, is_xrt):
     dataset = client.dataset(DATASET_NAME)
     logging.debug('Writing bigquery data to table %s in dataset %s',
                   TABLE_NAME, dataset)
-    table = bigquery.Table(name=TABLE_NAME, dataset=dataset)
-    table.reload()
-    return table.insert_data(row)
+    table_ref = bigquery.TableReference(dataset_ref=dataset, table_id=TABLE_NAME)
+    client.get_table(table_ref)
+    return client.create_rows(table, row)
```
Comment (liyanhui1228) [inline integration_tests/testsuite/deploy_app.py]: `table` should be `table_ref`?

Comment (nkubala) [reply integration_tests/testsuite/deploy_app.py]: yep good catch

File: integration_tests/testsuite/deploy_app.py [hunk 2]
```diff
@@ -72,9 +72,10 @@ def _record_latency_to_bigquery(deploy_latency, language, is_xrt):
     dataset = client.dataset(DATASET_NAME)
     logging.debug('Writing bigquery data to table %s in dataset %s',
                   TABLE_NAME, dataset)
-    table = bigquery.Table(name=TABLE_NAME, dataset=dataset)
-    table.reload()
-    return table.insert_data(row)
+    table_ref = bigquery.TableReference(dataset_ref=dataset,
+                                        table_id=TABLE_NAME)
+    client.get_table(table_ref)
+    return client.create_rows(table_ref, row)
```
Comment (liyanhui1228) [inline integration_tests/testsuite/deploy_app.py]: Just checked the [documentation](https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/bigquery/google/cloud/bigquery/client.py#L1023) that if using `table_ref`, then `selected_fields` param is required when calling `create_rows`. So might just passed in a `Table` object:

```
table = client.get_table(table_ref)
return client.create_rows(table, row)
```
I guess you probably wanted to pass in the `Table` object but forgot to create it when calling `client.get_table` :)

Comment (nkubala) [reply integration_tests/testsuite/deploy_app.py]: Ah, yeah that's right. Sorry I thought I had tested this correctly but never actually checked the table I wrote to :fearful:

**Comment by dlorenc** [comment]
@nkubala feel free to merge, travis passed but looks like it's having trouble updating the status.

**Pull request merged**